share/build/buildlib.kokkos

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+
+import os, sys, argparse, logging
+
+from standard_script_setup import *
+from CIME import utils
+from CIME.utils import expect, run_bld_cmd_ensure_logging, run_cmd_no_fail, run_cmd
+from CIME.case import Case
+from CIME.build import get_standard_makefile_args
+
+logger = logging.getLogger(__name__)
+
+
+def parse_command_line(args, description):
+    ###############################################################################
+    parser = argparse.ArgumentParser(
+        usage="""\n{0} [--debug]
+OR
+{0} --verbose
+OR
+{0} --help
+
+\033[1mEXAMPLES:\033[0m
+    \033[1;32m# Run \033[0m
+    > {0}
+""".format(
+            os.path.basename(args[0])
+        ),
+        description=description,
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+
+    utils.setup_standard_logging_options(parser)
+
+    parser.add_argument("buildroot", help="build path root")
+
+    parser.add_argument("installpath", help="install path ")
+
+    parser.add_argument(
+        "caseroot", nargs="?", default=os.getcwd(), help="Case directory to build"
+    )
+
+    args = utils.parse_args_and_handle_standard_logging_options(args, parser)
+
+    return args.buildroot, args.installpath, args.caseroot
+
+
+###############################################################################
+def buildlib(bldroot, installpath, case):
+    ###############################################################################
+    srcroot = case.get_value("SRCROOT")
+    kokkos_dir = os.path.join(srcroot, "externals", "kokkos")
+    expect(os.path.isdir(kokkos_dir), "Missing kokkos submodule")
+
+    # We want to get the compiler and kokkos_options from Macros.make
+    # (generated from cmake_macros), but we want to otherwise
+    # let kokkos control flags
+    make_args = get_standard_makefile_args(case, shared_lib=True)
```

Comment:
I'm thinking of a CIME util along the line of
```
def get_cmake_option (...):
  content = ""
  for f in files:
    content += f"include ({f})\n"

  f = open("CMakeLists.txt","w")
  f.write(content)
  f.write(f'message ("{var} = ${{{var}}}")')
  f.close()
  return run_cmd_no_fail ("cmake . | grep {var}")
```

Reply:
@bartgol , yes, this is a very unique use case where need to query a CMake option that is computed by the macros, not the case settings. The case control system will have already made the Macros.make file to support the sharedlib builds, so I think this solution is decent since it doesn't require any extra file creation.